forge/db/views/BrokerCredentials.js

```diff
@@ -11,7 +11,8 @@ module.exports = function (app) {
             protocolVersion: { type: 'number' },
             ssl: { type: 'boolean' },
             verifySSL: { type: 'boolean' },
-            clientId: { type: 'string' }
+            clientId: { type: 'string' },
+            topicPrefix: { type: 'string' }
```

Comment:
I personally think this is too restrictive. 

The original discussion in slack was to support multiple topics since brokers who deny subscription to `#` would mean a user can only subscribe to a single 2nd level topic e.g. `press/#` and thus not able to see topics in `body/#` or  `paint/#`

I have no issue with supporting a single custom topic for the first iteration however by setting the API in stone and to a single `string` prevents an easy upgrade path in the future (without having to modify the API signature)

If you have no issue with redefining the API at a future time, I will approve.

Reply:
> We may wish to include a topic filter on the broker configuration.

> Will scope this to adding a config option in the Broker settings to define the top-level scope that you wish to subscribe to for traffic, defaulting to # (for now)

This meets what was asked for

Reply:
The issue raised did not reflect the requirement or discussion in slack.  

Your PR of course does meet the issue but that does not detract from the shortcomings.